lib/resty/acme/autossl.lua

```diff
@@ -234,6 +251,22 @@ function AUTOSSL.update_cert(data)
 
   err = update_cert_handler(data)
 
+  local failure_count_key = certificate_failure_count_prefix .. ":" .. data.domain
+  if err then
+    local count_storage, _ = AUTOSSL.storage:get(failure_count_key)
+    local count = (count_storage or 0) + 1
+    AUTOSSL.storage:set(failure_count_key, count)
+
+    if failure_cooloff_callback then
+      local cooloff = failure_cooloff_callback(domain, count)
+      AUTOSSL.storage:add(failure_lock_key, "1", cooloff)
```

Comment:
One gotcha when developing the kong acme plugin is that you can make the value lock useful by storing the unix timestamp when cooloff is passed, so that when reading you can also say in log that cooloff will pass in x seconds for example.